### PR TITLE
Update texture_atlas example with different padding and sampling

### LINKED ISSUE #9522: Expand the texture atlas builder example to demonstrate more features
>  It might also be good to expand the texture atlas builder example in another PR.
> We could generate atlases with and without padding and render the textures with and without anti-aliasing to demonstrate how different approaches can deal with bleeding issues etc.

_Originally posted by @ickshonpe in https://github.com/bevyengine/bevy/issues/9494#issuecomment-1685908830_

### PR DESCRIPTION
**Pull request by antoniacobaeus**
# Objective

- Expand the texture_atlas example with padding and show how it can resolve sprite bleeding for different types of sampling. 
- Fixes #9522
## Solution
Updated the texture_atlas example by adding 4 different texture atlases:
1. linear, no padding
2. linear, padding
3. nearest neighbor, no padding
4. nearest neighbor, padding

Now renders one padded and one unpadded texture atlas, and the same upscaled sprite from each of the new texture atlases. See the screenshot below (taken on 1080p monitor).

Screenshot from 2023-10-10 08-37-43

**From left->right:** linear no padding, nearest no padding, linear padding, nearest padding.

---

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]
Nice stuff! Good docs, simple and clean code.

**Review by ickshonpe — CHANGES_REQUESTED** [review]
Two requests: 
* I have to resize the window to view all of the images. Everything should fit into the default window size.
* It would be good to have a label above each of the piggy shopkeeper sprites with the details of the filtering and padding that has been applied to the sprite.

Otherwise looks good, does exactly what we wanted.

**Comment by antoniacobaeus** [comment]
Added the requests mentioned above. The example should now fit in the default window size, and labels are added to display the different settings, see image below.

Screenshot from 2023-10-13 17-29-0525

*Since it seems like text2d is broken on main I had to merge with #9708 to test the labels. I manage to get the text to show, but got some artifacts which shouldn't be present when text2d works properly.*

File: examples/2d/texture_atlas.rs [hunk 1]
```diff
@@ -49,10 +56,184 @@ fn setup(
...
+    );
+
+    // Nearest, right
+    create_label(
+        &mut commands,
+        (350.0, 280.0, 0.0),
+        "Nearest",
+        text_style_smaller,
+    );
+}
```
Comment (ickshonpe) [inline examples/2d/texture_atlas.rs:223]: The repeated `create_label` calls could be refactored using a closure or loop maybe. It's not blocking though.

File: examples/2d/texture_atlas.rs [hunk 2]
```diff
@@ -49,10 +56,184 @@ fn setup(
...
+        atlas_linear_padded_handle,
+    );
+
+    // nearest, padding
+    create_sprite_from_atlas(
+        &mut commands,
+        (350.0, 170.0, 0.0),
+        vendor_index,
+        atlas_nearest_padded_handle,
+    );
```
Comment (ickshonpe) [inline examples/2d/texture_atlas.rs:165]: Same comment as for the labels below, this section could be refactored a bit.

File: examples/2d/texture_atlas.rs [hunk 3]
```diff
@@ -1,11 +1,17 @@
-//! In this example we generate a new texture atlas (sprite sheet) from a folder containing
+//! In this example we generate 4 texture atlases (sprite sheets) from a folder containing
```
Comment (ickshonpe) [inline examples/2d/texture_atlas.rs:1]:
```suggestion
//! In this example we generate four texture atlases (sprite sheets) from a folder containing
```

Word number seems more readable here

File: examples/2d/texture_atlas.rs [hunk 4]
```diff
@@ -1,11 +1,17 @@
...
 //! individual sprites.
+//!
+//! The texture atlases are generated with different padding and sampling to demonstrate the
+//! effect of these settings, and how bleeding issues can be resolved by padding the sprites.
+//!
+//! Only one padded and one unpadded texture atlas are rendered to the screen.
+//! An upscaled sprite from each of the 4 atlases are rendered to the screen.
```
Comment (ickshonpe) [inline examples/2d/texture_atlas.rs:8]:
```suggestion
//! An upscaled sprite from each of the four atlases are rendered to the screen.
```

**Review by ickshonpe — APPROVED** [review]
Sorry for not rereviewing earlier, the changes are exactly what I asked for. Looks great now. 
`Text2d` works again I think, so once you've merged with main this is ready to go.

**Comment by alice-i-cecile** [comment]
CI is failing: fix that up and I'll merge this in for you <3

**Pull request merged**